addon/array/stateful.js

```diff
@@ -14,16 +14,22 @@ import { fragmentDidDirty, fragmentDidReset } from '../states';
 
   @class StatefulArray
   @namespace MF
-  @extends Ember.ArrayProxy
+  @extends Ember.MutableArray
 */
-const StatefulArray = ArrayProxy.extend(Copyable, {
+const StatefulArray = EmberObject.extend(MutableArray, Copyable, {
   /**
     A reference to the array's owner record.
 
     @property owner
     @type {DS.Model}
   */
-  owner: null,
+  get owner() {
+    const owner = this.recordData.getFragmentOwner();
+    if (!owner) {
+      return null;
+    }
+    return this.store._internalModelForResource(owner.identifier).getRecord();
```

Comment:
you can just call `store.peekRecord(owner.identifier)`

Reply:
Looks like that was added in 3.29 https://github.com/emberjs/data/commit/3bc81a6ee6ba980e7201c7bb936cee635e706274

Reply:
@dwickern there is no 3.29, I think an alpha of 4.0 got tagged that way. Fairly sure we landed this for 3.28 let me check, but if not I can backport the capability to make it easier.

Reply:
hrm yeah, you need 4.0. I can backport that to 3.28 because otherwise you don't have a way of retrieving newly created records that don't have an ID yet.

Reply:
I don't think we need it backported. We can use `_internalModelForResource` for 3.8 and `peekRecord` on 4+.

Reply:
Was this resolved?

Reply:
I don't think it's blocking for 3.28, maybe something to address for ember-data v4